src/styles/getTooltipStyles.js

```diff
@@ -0,0 +1,177 @@
+        return roundToNearestMultipleOfFour(GUTTER_WIDTH - tooltipLeftEdge);
+    }
+
+    if (tooltipRightEdge > (windowWidth - GUTTER_WIDTH)) {
+        // Tooltip is in right gutter, shift left by a multiple of four.
+        return roundToNearestMultipleOfFour(windowWidth - GUTTER_WIDTH - tooltipRightEdge);
+    }
+
+    // Tooltip is not in the gutter, so no need to shift it horizontally
+    return 0;
+}
+
+/**
+ * Generate styles for the tooltip component.
+ *
+ * @param {Number} currentSize - The current size of the tooltip used in the scaling animation.
+ * @param {Number} windowWidth - The width of the window.
+ * @param {Number} xOffset - The distance between the left edge of the window
+ *                           and the left edge of the wrapped component.
+ * @param {Number} yOffset - The distance between the top edge of the window
+ *                           and the top edge of the wrapped component.
+ * @param {Number} componentWidth - The width of the wrapped component.
+ * @param {Number} componentHeight - The height of the wrapped component.
+ * @param {Number} tooltipWidth - The width of the tooltip itself.
+ * @param {Number} tooltipHeight - The height of the tooltip itself.
+ * @returns {Object}
+ */
+export default function getTooltipStyles(
+    currentSize,
+    windowWidth,
+    xOffset,
+    yOffset,
+    componentWidth,
+    componentHeight,
+    tooltipWidth,
+    tooltipHeight,
+) {
+    // Determine if the tooltip should display below the wrapped component.
+    // If a tooltip will try to render within GUTTER_WIDTH logical pixels of the top of the screen,
+    // we'll display it beneath its wrapped component rather than above it as usual.
+    const shouldShowBelow = (yOffset - tooltipHeight) < GUTTER_WIDTH;
+
+    // Determine if we need to shift the tooltip horizontally to prevent it
+    // from displaying too near to the edge of the screen.
+    const horizontalShift = computeHorizontalShift(windowWidth, xOffset, componentWidth, tooltipWidth);
+
+    const tooltipVerticalPadding = spacing.pv1;
+    const tooltipFontSize = variables.fontSizeSmall;
+
+    return {
+        animationStyle: {
+            transform: [{
+                scale: currentSize,
+            }],
+        },
+        tooltipWrapperStyle: {
+            position: 'absolute',
+            backgroundColor: themeColors.heading,
+            borderRadius: variables.componentBorderRadiusSmall,
+            ...tooltipVerticalPadding,
+            ...spacing.ph2,
+
+            // Shift the tooltip down (+) by...
+            //
+            //   component height
+            // + (pointer height - 1)
+            //
+            // OR
+            //
+            // Shift the tooltip up (-) by...
+            //   font height
+            // + top padding
+            // + bottom padding
+            // + upward shift of the pointer (pointer height)
+            // + (pointer height - 1)
+            //
```

Comment:
I think this comments could be more valuable if they did more than just repeat what we are doing in the code below and would maybe work better as a short sentence that reminds us why we are doing this.

>If we are showing the tooltip below we set the `top` value to be equal to the component height plus the height of the pointer minus 1 extra pixel for _____ to push things below the wrapped content.

>If we are showing the tooltip above then we must apply a negative `top` value and also take into account the size of the font, overall vertical padding and multiply the pointer height by 2 because...